src/fields.jl

```diff
@@ -14,13 +14,21 @@ A field datatype for 1D objects.
 
 $(TYPEDFIELDS)
 """
-struct Field1D{LX<:AbstractLocation} <: AbstractField
+struct Field1D{LX<:AbstractLocation, G} <: AbstractField
     "Array with the values of the field."
-    data :: Array
+    data::AbstractArray
```

Comment:
```suggestion
    data :: AbstractArray
```

Reply:
I think you want to use a type parameter here. Otherwise `Field` is not concretely type and could suffer from performance issues.

Reply:
> I think you want to use a type parameter here. Otherwise `Field` is not concretely type and could suffer from performance issues.

Or simply
```
data :: OffsetArray
```
?

Reply:
```julia
struct Field{L, G, A}
    data :: A
    grid :: G

    # Inner constructor
end
```